### PR TITLE
Baymed Phase 1, Part 2: Death, mob lists and bodyparts

### PR DESCRIPTION
**Pull request by NanakoAC**
Another PR full of boring nerd work, this ports the contents of bay's mob/death.dm
Also ports bay's organ defines, and converts our global mob lists to the GLOB.thing style

This PR is almost entirely composed of find and replace enmasse, so there's a lot of files changed, but very little functional change

### CODE REVIEW

File: code/game/objects/items/weapons/power_cells.dm [hunk 1]
```diff
@@ -71,7 +71,7 @@
 	desc = "A rechargable starch based power cell."
 	origin_tech = list(TECH_POWER = 1)
 	icon = 'icons/obj/power.dmi'
-	icon_state = "potato_cell"
+	icon_state = "potatBP_CELL"
```
Comment (mattroks101) [inline code/game/objects/items/weapons/power_cells.dm]: Mass replace fucked something up here.

File: code/modules/mob/living/carbon/human/human_helpers.dm [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #define HUMAN_EATING_NO_ISSUE		0
-#define HUMAN_EATING_NO_MOUTH		1
+#define HUMAN_EATING_NBP_MOUTH		1
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/human/human_helpers.dm]: Oh dear this went wrong too

File: code/modules/mob/living/carbon/human/species/species_attack.dm [hunk 3]
```diff
@@ -25,7 +25,7 @@
 		return 0
 
 	switch(zone)
-		if(BP_HEAD, "mouth", O_EYES)
+		if(BP_HEAD, "mouth", BP_EYES)
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/human/species/species_attack.dm]: This should be BP_MOUTH i think

File: code/modules/mob/living/carbon/human/species/xenomorphs/alien_species.dm [hunk 4]
```diff
@@ -148,11 +148,11 @@
 	deform =  'icons/mob/human_races/xenos/r_xenos_drone.dmi'
 
 	has_organ = list(
-		O_HEART =  /obj/item/organ/internal/heart,
-		O_BRAIN =  /obj/item/organ/internal/brain/xeno,
-		O_PLASMA = /obj/item/organ/internal/xenos/plasmavessel/drone,
+		BP_HEART =  /obj/item/organ/internal/heart,
+		BP_BRAIN =  /obj/item/organ/internal/brain/xeno,
+		BP_PLASMA = /obj/item/organ/internal/xenos/plasmavessel/drone,
 		O_ACID =   /obj/item/organ/internal/xenos/acidgland/drone,
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/human/species/xenomorphs/alien_species.dm]: some organs here that missed the sweep

File: code/modules/mob/living/carbon/human/unarmed_attack.dm [hunk 5]
```diff
@@ -149,7 +149,7 @@ var/global/list/sparring_attack_cache = list()
 
 	if(!target.lying)
 		switch(zone)
-			if(BP_HEAD, "mouth", O_EYES)
+			if(BP_HEAD, "mouth", BP_EYES)
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/human/unarmed_attack.dm]: mouth here needs changed

File: code/modules/mob/living/living.dm [hunk 6]
```diff
@@ -369,7 +369,7 @@ default behaviour is:
 /mob/living/proc/get_organ_target()
 	var/mob/shooter = src
 	var/t = shooter:targeted_organ
-	if(t in list(O_EYES, "mouth"))
+	if(t in list(BP_EYES, "mouth"))
```
Comment (NanakoAC) [inline code/modules/mob/living/living.dm]: mouth

**Comment by NanakoAC** [comment]
all changes done

File: code/__defines/damage_organs.dm [hunk 7]
```diff
@@ -25,40 +25,74 @@
...
-#define BP_ALL list(BP_CHEST, BP_GROIN, BP_HEAD, BP_L_ARM, BP_R_ARM, BP_L_LEG, BP_R_LEG)
+#define BP_ARMS list(BP_R_ARM, BP_L_ARM)
+#define BP_ALL_LIMBS list(BP_CHEST, BP_GROIN, BP_HEAD, BP_L_ARM, BP_R_ARM, BP_L_LEG, BP_R_LEG)
+#define BP_BY_DEPTH list(BP_HEAD, BP_L_ARM, BP_R_ARM, BP_L_LEG, BP_R_LEG, BP_GROIN, BP_CHEST)
+
+// Prosthetic helpers.
+#define BP_IS_ROBOTIC(org)  (org.status & ORGAN_ROBOTIC)
+#define BP_IS_ASSISTED(org) (org.status & ORGAN_ASSISTED)
+#define BP_IS_BRITTLE(org)  (org.status & ORGAN_BRITTLE)
+#define BP_IS_CRYSTAL(org)  (org.status & ORGAN_CRYSTAL)
```
Comment (LudwigVonChesterfield) [inline code/__defines/damage_organs.dm]: This might never get used as it's only current use in Bay's special snowflake race. Much like the define above.

Comment (NanakoAC) [reply code/__defines/damage_organs.dm]: don't worry about that, i might find a use in the future x

**Pull request merged**